### DISCUSSION TITLE
Use Hasura instead of Primsa/sqlite/postgres?

### DISCUSSION THREAD

**Discussion by cj** [blitz-js/blitz#2319] [Q&A]
Hi is it possible to use Hasura instead of Primsa/sqlite/postgres?

I see I can change most everything to use Hasura instead of `db`, but it seems like I don't have access to override the way sessions are set.  Is there a simple way I can hack it to allow my own custom session handler?

Cheers!

**Comment by flybayer** [comment]
Yes you can do that! See https://blitzjs.com/docs/session-management#customize-session-persistence-and-database-access. You can see an example using graphql here: https://github.com/blitz-js/blitz/blob/canary/examples/fauna/blitz.config.js#L24

**Reply by cj** [reply]
@flybayer Thank you!  FYI - it appears the link in the docs here https://blitzjs.com/docs/session-management#customize-session-persistence-and-database-access do not work.

**Reply by cj** [reply]
@flybayer I removed all the places where it uses db, but I still get this error:

```
error - ./db/index.ts:2:0
Module not found: Can't resolve '@prisma/client'
  1 | import { enhancePrisma } from "blitz"
> 2 | import { PrismaClient } from "@prisma/client"
  3 |
  4 | const EnhancedPrisma = enhancePrisma(PrismaClient)
```

So it looks like there is a hard requirement for prisma even if you are not using it.

What do you think of the idea of having a `--no-database` option when running `blitz new`?

**Reply by flybayer** [reply]
You need to either remove `db/index.ts` or remove prisma from that file.

And yeah, we can add more options to blitz new. PRs welcome!

And I just fixed the link in the docs. Should be live within a minute or two.

**Reply by cj** [reply]
@flybayer I had removed the db folder completely and anything that uses db, but I was still getting that error.  I think there is still something internally to Blitjs that is using it.
![CleanShot 2021-05-12 at 15 51 18](https://user-images.githubusercontent.com/1819/118042649-f6335880-b339-11eb-8e1e-259d6408ca11.png)

**Reply by flybayer** [reply]
Ok, first try deleting `.next` and `.blitz`. If issue still persists, then open a bug report with exact steps to reproduce (repo or something ideally)

**Reply by cj** [reply]
Will do - thank you!
